react-graph/src/Step.tsx

```diff
@@ -0,0 +1,112 @@
+// ******************************************************************************************************
+//  Step.tsx - Gbtc
+//
+//  Copyright � 2020, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  12/13/2022 - A. Hagemeyer
+//       Generated original version of source code.
+//
+// ******************************************************************************************************
+
+import * as React from 'react'
+
+interface IProps {
+    steps:
+    {
+        short: string,
+        long: string,
+        id: string | number
+    }[],
+    activeStep: string | number,
+    height: number,
+    width: number
+}
+
+function Step(props: IProps) {
+    let met = false;
```

Comment:
Correct me if I am wrong, but wouldn't this declaration being here cause issues on re-renders? Let's say the step changes without re-rendering this component, wouldn't met start as true, messing up the function getSteps()?

Reply:
a change in a `prop` always triggers a re-render so this should not be an issue